### ISSUE TITLE
Minimize the use of `any` types

### ISSUE THREAD

**Issue by mcmire** [MetaMask/core#751]
Right now TypeScript ESLint's `no-explicit-any` rule is turned off in our global ESLint config: https://github.com/MetaMask/eslint-config/blob/ec27156931a0cda1b34ff46937b8296742dcb1c0/packages/typescript/src/index.js#L39. While changing this setting globally is a different issue, we should consider enabling it just for `controllers`. We should be using `any` as little as possible, because using it indicates that something is not typed properly, whether it is a dependency or our own code.

Enabling this rule produces 335 violations across ~15 files.

This issue is partially blocked by #728.

**Comment by BboyAkers** [comment]
@mcmire would it help to reactor the `any` keywords into `unknown`? It's more type safe than `any`

**Comment by mcmire** [comment]
@BboyAkers We could, and that would certainly highlight the fact that we are using `any`. But if we are changing types, it would be even better to see if we can figure out more accurate types to use rather than defaulting to `unknown`.

**Comment by BboyAkers** [comment]
> @BboyAkers We could, and that would certainly highlight the fact that we are using `any`. But if we are changing types, it would be even better to see if we can figure out more accurate types to use rather than defaulting to `unknown`.

True, true, true! I can drive into a couple of places and start investigating 🙂

**Comment by MajorLift** [comment]
794 errors for `@typescript-eslint/no-explicit-any` are being reported in core as of today.

**Comment by MajorLift** [comment]
- Fixed by https://github.com/MetaMask/core/pull/3660
- Superseded by https://github.com/MetaMask/core/issues/3706
